public/ipython/tree/js/notebooklist.js

```diff
@@ -274,6 +275,14 @@ define([
     };
 
 
+    NotebookList.prototype.read_only_url = function(url) {
```

Comment:
Since it is attached to the notebook list instance, you don't think it's fair?

Or you propose to have a static function that would take both `url` and `viewer` as params?